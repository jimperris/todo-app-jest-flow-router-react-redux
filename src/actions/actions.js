//@flow
import type {TodoFilterActions} from '../types/basic'
var num: number = 0
export const addTodo = (title: string): TodoFilterActions => {
  return {
    type: 'ADD_TODO',
    id: num++,
    title: title
  }
}
export const removeTodo = (id: number): TodoFilterActions => {
  return {
    type: 'REMOVE_TODO',
    id: id
  }
}

export const setVisibilityFilter = (filter: string) => {
  return {
    type: 'SET_VISIBILITY_FILTER',
    filter: filter
  }
}

export const toggleTodo = (id: number) => {
  return {
    type: 'TOGGLE_TODO',
    id: id
  }
}

export const VisibilityFilters = {
  SHOW_ALL: 'SHOW_ALL',
  SHOW_COMPLETED: 'SHOW_COMPLETED',
  SHOW_IN_PROGRESS: 'SHOW_IN_PROGRESS'
}
