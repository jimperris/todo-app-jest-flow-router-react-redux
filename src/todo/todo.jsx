/* @flow */
import type {Todos} from '../types/basic'
import React from 'react'
import { Link } from 'react-router-dom'

export const findTodo = (todos: Todos, id: number) => {
  for(var i = 0; i < todos.length; i++) {
    if(todos[i].id === id) {
      return todos[i]
    }
  }
  console.log('ERROR: TODO NOT FOUND')
  return {}
}
const TodoComp = (
  id: number, todos: Todos,
  toggleTodo: (id: number) => void, removeTodo: (id: number) => void
) => {
  let todo = findTodo(todos, id)
  return (
    <div>
      <Link to='/'>Home</Link>
      <h2 style={{border: '1px solid black'}}
        onClick={() => toggleTodo(todo.id)}
      >
        {todo.title}
      </h2>
      <h4>{todo.complete ? 'complete' : 'in progress'}</h4>
      <Link to='/' onClick={() => removeTodo(todo.id)}>
        Remove
      </Link>
    </div>
  )
}

export default TodoComp
