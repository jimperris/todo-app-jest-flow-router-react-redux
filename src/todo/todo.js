//@flow
import Todo from './todo.jsx'
import type {State, Dispatch} from '../types/basic'
import {connect} from 'react-redux'
import * as actions from '../actions/actions'
export const mapStateToProps = (state: State) => {
  return {
    todos: state.todos
  }
}
export const mapDispatchToProps = (dispatch: Dispatch) => ({
  removeTodo: (id: number) => dispatch(actions.removeTodo(id)),
  toggleTodo: (id: number) => dispatch(actions.toggleTodo(id))
})

export const TodoContainer = connect(mapStateToProps, mapDispatchToProps)(Todo)
