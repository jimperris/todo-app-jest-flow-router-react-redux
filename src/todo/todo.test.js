import Todo from './todo.jsx'
import { mount } from 'enzyme'
import React from 'react'
import * as actions from '../actions/actions'
import {findTodo} from './todo.jsx'
import {mapStateToProps, mapDispatchToProps} from './todo'
import {BrowserRouter as Router} from 'react-router-dom'
describe('maps', () => {
  it('mapStateToProps', () => {
    expect(mapStateToProps({todos: [{title: 'hairo'}]})).toEqual({todos: [{title: 'hairo'}]})
  })
  it('mapDispatchToProps', () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).removeTodo('3')
    expect(dispatch.mock.calls[0][0]).toEqual(actions.removeTodo('3'))
  })
})

describe('findTodo function', () => {
  let todo = findTodo({todos: [{id: 2}, {title: 'ooooof', id: '14'}], id: '14'})
  expect(todo).toEqual({title: 'ooooof', id: '14'})
})

describe('post', () => {
  const wrap = mount(
    <Router>
      <Todo id={0} todos={[{title: 'name', id: 0, complete: false}]} />
    </Router>
  )
  it('should render and keep props', () => {
    expect(wrap.exists()).toBeTruthy()
  })
  it('should match snapshot', () => {
    expect(wrap).toMatchSnapshot()
  })
})
