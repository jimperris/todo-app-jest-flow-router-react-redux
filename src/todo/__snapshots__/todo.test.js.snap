// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post should match snapshot 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <h2
    onClick={[Function]}
    style={
      Object {
        "border": "1px solid black",
      }
    }
  >
    name
  </h2>
  <h4>
    in progress
  </h4>
  <a
    href="/"
    onClick={[Function]}
  >
    Remove
  </a>
</div>
`;
